### DISCUSSION TITLE
Middleware support? (e.g. apollo-links)

### DISCUSSION THREAD

**Discussion by nuragic** [nearform/graphql-hooks#761] [Ideas]
Just opening this discussion to see how many people would be interested to have this. All opinions, use cases etc are welcome!

**Comment by zanettin** [comment]
Thanks for starting this 🙏  I am in general a fan of having a middleware/plugin option. it makes the core lib extensible w/o bloating it for everybody. so you're free to add what you need and you'd open the door for any 3rd party plugins as well. when it comes to use cases we could imaging pretty everything that apollo has. request transforms, response transforms, error handlers, retry logics, persistence logic, etc

**Comment by salmanm** [comment]
Yea I always wanted a way for us to be able to transform request/response and stuff with gql-hooks. Doing that with custom axios instance and interceptors is a bit ugly and limiting.
So I'd be in favor for the middleware support.

**Comment by simoneb** [comment]
I'm in favor of this, especially if it's a good approach to implement #759

**Comment by nuragic** [comment]
Ok, seems that this discussion is settled. I’m going to create a new issue for adding middleware support. Thanks!
